### ISSUE TITLE
A Gamification Bot

### ISSUE THREAD

**Issue by aps120797** [probot/ideas#55]
The concept behind the bot is **[Gamification](https://en.wikipedia.org/wiki/Gamification)**.

In simple terms, gamification will encourage more users to contribute and work with the repositories.

**How will this be implemented?**
This could be implemented by using bot, built on Probot and possibly, `.github/config.yml` for storing points. For example, a bot will be maintain a points depending on the user contribution (E.g., +2 for an issue, +5 for a PR, +10 for a positive reply from someone (like Thank you), etc.). These points will be accumulated and the users will be awarded achievements for their contribution, (Example. users with 100+ points will receive an achievement MergerKing, 200+ points will be awarded PullEmperor, etc).
If GitHub allows, these achievements could be added to the user's GitHub profile (I don't know how to do it).
And many other additional awards and achievements could be awarded! 

The major reason why I think this will an interesting bot is because it will improve competency and improve user engagement.

This is just an idea, possibly contributors, (gamers), and Bot Builders can help improve my idea.

Here's a YouTube video which easily explains what Gamification is: [What is Gamification?](https://www.youtube.com/watch?v=BqyvUvxOx0M)

If you feel this idea is cool, please give a 👍 or 🎉 and let the maintainers know if the idea is interesting.
Also, if this idea is cool, can this idea be reserved for GSoC?

**Comment by knrt10** [comment]
@aps120797 this is really a nice idea!!

**Comment by itaditya** [comment]
@aps120797 [sendgrid](https://github.com/sendgrid?page=2) did something just like this for maintaining a leaderboard during hacktoberfest.

**Comment by aps120797** [comment]
@itaditya , Yeah, they made a leader-board of the organisations getting most contribution from the users participating in hacktoberfest, i remember, FCC topped that list. 😄

**Comment by aps120797** [comment]
Yeah, but this idea is a little different.. Here.. points ad achievements will be awarded to users for there contributions..

**Comment by itaditya** [comment]
I just said it, to give some validation to your idea. I meant that this may as well be a common requirement for many.

**Comment by gr2m** [comment]
I think I’d rather not reserve it for GSoC, but find a team of people passionate about the idea and then collaborate on it. I personally have reservations against gamification in open source communities, or volunteer communities in general, but I love experimentation and see what comes out of this :)

@aps120797 could you please remove the mention of the game from your comment. It does not add relevant information to the discussion. Thanks :)

**Comment by aps120797** [comment]
@gr2m Removed the mention of the game from the comment. Thanks. 😄

**Comment by rsarky** [comment]
@aps120797 I kid you not I was thinking about this a while back :astonished: 
This is an awesome idea in principle but I think what might happen is people might forget the spirit of open source code in order to get more points. This means stuff like spamming comments or issues to get more points and other stuff along those lines.

**Comment by rsarky** [comment]
@aps120797 I would love to collaborate with you on this sometime if you want?

**Comment by aps120797** [comment]
@rsarky , yeah, I was also worried about the spams. However, I think, if it is implemented in a proper, this could be a cool experiment. I will let you know when I will start working on it. Currently, waiting for more users to show interest. 😄

**Comment by jainkeval** [comment]
@aps120797 I would also like to collaborate with you!

**Comment by choznerol** [comment]
I like the idea! Adding the achievement to user's GitHub pages sounds like a big project, but this reminds me of the [embedded real-time board of sponsor avatars](https://github.com/vuejs/vue#contribution) in some big projects. What do you think if the bot could maintaining a leaderboard on the README?

**Comment by aps120797** [comment]
> Adding the achievement to user's GitHub pages sounds like a big project

Yeah, it's a big deal to implement this. But if it's done, it will be really cool. I am not sure if GitHub allows adding these details to users profile. 😄 
> [embedded real-time board of sponsor avatars](https://github.com/vuejs/vue#contribution)

This is really cool. Thanks for sharing! 😄
@itaditya too shared a similar idea of maintaining a leader-board. Maintaining a leader-board in `readme.md` is the best that we can do. 😄 

Project's taking shape. Any suggestions what should I name the project people?

**Comment by itaditya** [comment]
>  Any suggestions what should I name the project people?

* Golden Eye
* Uatu

**Comment by itaditya** [comment]
* Uatu

**Comment by haxzie** [comment]
Hey @abhijeetps ,
Is the project ready? I was planning to build something similar for a competition within our students community. If it already exists, that'd be cool, if it's under WIP I'd be happy to contribute to it.

**Comment by DarraghClarke** [comment]
I'm not entirely sure if this is what the Original creator had in mind, but I've given an effort to implementing a small bot along these lines

[Heres the repo for the bot](https://github.com/DarraghClarke/QMebhBot)
